subworkflows/local/bamfiltering.nf

```diff
@@ -69,61 +71,29 @@ workflow FILTER_BAM {
-    }
-
-    if ( ( params.run_metagenomics && params.metagenomics_input == 'unmapped' ) && params.preprocessing_skippairmerging ) {
-        ch_paired_fastq_for_cat = SAMTOOLS_FASTQ_UNMAPPED.out.fastq
-                                    .mix(SAMTOOLS_FASTQ_UNMAPPED.out.singleton)
-                                    .mix(SAMTOOLS_FASTQ_UNMAPPED.out.other)
-                                    .groupTuple()
-                                    .map {
-                                        meta, fastqs ->
-                                            def meta_new = meta.clone()
-                                            meta_new['single_end'] = true
-                                        [ meta_new, fastqs.flatten() ]
-                                    }
-        CAT_FASTQ_UNMAPPED ( ch_paired_fastq_for_cat )
-    }
-
-    // TODO: see request https://github.com/nf-core/eager/issues/945
-    if ( ( params.run_metagenomics && ( params.metagenomics_input == 'mapped' || params.metagenomics_input == 'all' ) ) && params.preprocessing_skippairmerging ) {
-        ch_paired_fastq_for_cat = SAMTOOLS_FASTQ_MAPPED.out.fastq
-                                    .mix(SAMTOOLS_FASTQ_MAPPED.out.singleton)
-                                    .mix(SAMTOOLS_FASTQ_MAPPED.out.other)
-                                    .groupTuple()
-                                    .map {
-                                        meta, fastqs ->
-                                            def meta_new = meta.clone()
-                                            meta_new['single_end'] = true
-                                        [ meta_new, fastqs.flatten() ]
-                                    }
-        CAT_FASTQ_MAPPED ( ch_paired_fastq_for_cat )
-    }
-
-    // Routing for metagenomic screening -> first accounting for paired-end mapping, then merged mapping, then no metagenomics
-    if ( ( params.run_metagenomics && params.metagenomics_input == 'unmapped' ) && params.preprocessing_skippairmerging ) {
-        ch_fastq_for_metagenomics = CAT_FASTQ_UNMAPPED.out.reads
-    } else if ( ( params.run_metagenomics && ( params.metagenomics_input == 'mapped' || params.metagenomics_input == 'all' ) ) && params.preprocessing_skippairmerging ) {
-        ch_fastq_for_metagenomics = CAT_FASTQ_MAPPED.out.reads
-    } else if ( params.run_metagenomics && params.metagenomics_input == 'unmapped' ) {
-        ch_fastq_for_metagenomics = SAMTOOLS_FASTQ_UNMAPPED.out.other
-    } else if ( params.run_metagenomics && ( params.metagenomics_input == 'mapped' || params.metagenomics_input == 'all' )) {
-        ch_fastq_for_metagenomics = SAMTOOLS_FASTQ_MAPPED.out.other
-    } else if ( !params.run_metagenomics ) {
+    //    No length/quality filtering applies to metagenomic bam files (could be extension)
+    //    All bam -> fastq filtering options (-F 4, -f 4 or none) will be dealt with within modules.config
```

Comment:
@TCLamnidis please take a quick look -- i restructured everything into a single call, since the modules.config is easily robust enough to deal with the splitting. and it allows for much cleaner code in this section which should help going forward

Reply:
or @jfy133 who i think wrote the initial implementation...

Reply:
I think that makes it much easier! Nice one :)